### PR TITLE
Linux distros jobs on PRs + Fix linking issues on some platforms

### LINKED ISSUE #2100: Trigger Linux-Distributions jobs on PRs
##### **Is your feature request related to a problem?**
I noticed that the "Nightly - Linux Distributions" builds is failing on the `main` branch after merging #2090. However, in that PR all the CI checks were passing.

##### **Describe the solution you would like**
It would be good to also trigger the "Linux Distributions" builds in PRs, so that we do not end up breaking something in `mian` without noticing in the PRs.

We would just need to change `.github/workflows)/nightly_Linux_distributions.yml` so that it runs on PRs.

Furthermore, we should fix the current issue on `main`. It is complaining about undefined references to `std::filesystem`:

```bash
[100%] Linking CXX executable ../bin/exiv2
CMakeFiles/exiv2.dir/actions.cpp.o: In function `(anonymous namespace)::metacopy(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, bool)':
actions.cpp:(.text+0xa4e7): undefined reference to `std::filesystem::__cxx11::path::_M_split_cmpts()'
actions.cpp:(.text+0xa4f7): undefined reference to `std::filesystem::temp_directory_path[abi:cxx11]()'
actions.cpp:(.text+0xa50f): undefined reference to `std::filesystem::__cxx11::path::has_root_directory() cons
...
```

Something like [this](https://stackoverflow.com/a/54290445/1977218) should work.

### PR DESCRIPTION
**Pull request by piponazo**
Fixes #2100

### CODE REVIEW

**Comment by neheb** [comment]
Not quite correct.

For GCC 8 and clang 7-8, this is needed. GCC9 and clang9 and above do not. Well, the libc++ versions actually.

GCC7 supports std::experimental::filesystem which is basically boost::filesystem. I wouldn't bother with it though.

**Comment by piponazo** [comment]
Thanks for the detailed review! I'll be more explicit about those checks in the CMake code then.

**Comment by neheb** [comment]
This is how I do it elsewhere: https://github.com/gerbera/gerbera/pull/1868/commits/143217b220a2f46a4385143155296e6d328c32e7

It's not cmake but it's easy to translate I think.

**Comment by piponazo** [comment]
> This is how I do it elsewhere: [gerbera/gerbera@143217b#diff-30d8f6be6320feeacf686be94f48c70869b52630e01ea625f0f15adc0d57c3e4R16](https://github.com/gerbera/gerbera/commit/143217b220a2f46a4385143155296e6d328c32e7#diff-30d8f6be6320feeacf686be94f48c70869b52630e01ea625f0f15adc0d57c3e4R16)
> It's not cmake but it's easy to translate I think.

Just for the sake of sharing: There is a long discussion about this topic in a CMake gitlab issue:
https://gitlab.kitware.com/cmake/cmake/-/issues/17834

It seems difficult to come up with something very robust (for all possible cases). I'll try to come up with something good enough for the platforms we support in our CI infrastructure.

**Comment by neheb** [comment]
I forgot to mention one thing. On 32-bit OS, GCC 8.3 has a bug where std::filesystem does not work with larger than 2GB files. This was fixed with 8.4 I don't know if that's relevant. Debian 10 comes with GCC 8.3.

**Comment by piponazo** [comment]
> I forgot to mention one thing. On 32-bit OS, GCC 8.3 has a bug where std::filesystem does not work with larger than 2GB files. This was fixed with 8.4 I don't know if that's relevant. Debian 10 comes with GCC 8.3.

Definitely it is good to know. I'm adding this link for future reference in case this bites us in the future:
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=91947

File: cmake/FindFilesystem.cmake [hunk 1]
```diff
@@ -0,0 +1,247 @@
+# Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
+# file Copyright.txt or https://cmake.org/licensing for details.
+
+#[=======================================================================[.rst:
+
```
Comment (hassec) [inline cmake/FindFilesystem.cmake:5]: could we just add a brief comment with a link to where this was taken from? Just so if anything breaks one can check upstream for a potential fix?

**Review by hassec — CHANGES_REQUESTED** [review]

File: cmake/FindFilesystem.cmake (hunk 1 above)
Comment (piponazo) [reply cmake/FindFilesystem.cmake:5]: The thing is that I do not know the original author 😅 

I took this one from the link I shared previously from the CMake Issue:
https://gitlab.kitware.com/cmake/cmake/-/issues/17834

In there, somebody comments about this file existing at least in 2 repositories:
https://github.com/lethal-guitar/RigelEngine/blob/master/cmake/Modules/FindFilesystem.cmake
https://github.com/vector-of-bool/CMakeCM/blob/master/modules/FindFilesystem.cmake

Would it be enough to just have it here (in the PR discussion)? If something ever happen here, git blame should help to obtain more information.

Comment (hassec) [reply cmake/FindFilesystem.cmake:5]: Since this file shouldn't really change, I guess that's fine :+1:

**Review by hassec — APPROVED** [review]

**Pull request merged**